qiskit_experiments/framework/experiment_data.py

```diff
@@ -938,25 +959,33 @@ def _add_result_data(self, result: Result) -> None:
                 if hasattr(expr_result, "meas_return"):
                     data["meas_return"] = expr_result.meas_return
                 self._result_data.append(data)
+        self.end_datetime = datetime.now()
 
     def _retrieve_data(self):
         """Retrieve job data if missing experiment data."""
-        if self._result_data or not self._backend:
-            return
+        # if self._result_data or not self._backend:
+        #     return
```

Comment:
Remove commented code lines

Reply:
See the above discussion with @coruscating. We should decide on what to do here.